sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -402,16 +404,20 @@ private void initialize(SchedulerDriver driver) throws InterruptedException {
         initializeGlobals(driver);
         initializeDeploymentPlanManager();
         initializeRecoveryPlanManager();
+        initializePlanCoordinator();
         initializeResources();
         DcosCertInstaller.installCertificate(System.getenv("JAVA_HOME"));
-        final List<PlanManager> planManagers = Arrays.asList(
-                deploymentPlanManager,
-                recoveryPlanManager);
-        planCoordinator = new DefaultPlanCoordinator(planManagers, planScheduler);
         planCoordinator.subscribe(this);
         LOGGER.info("Done initializing.");
     }
 
+    private Collection<PlanManager> getOtherPlanManagers() {
+        return plans.stream()
+                .filter(plan -> !plan.getName().equals(DEPLOY_PLAN_NAME))
```

Comment:
Maybe add an `isDeployPlan(Plan)` to `PlanUtils`, replacing this check and a similar one in `initializeDeploymentPlanManager()`.